### ISSUE TITLE
[Beta] ScheduleInformation#getScheduleItems returns array not ScheduleItem[]

### ISSUE THREAD

**Issue by ghost** [microsoftgraph/msgraph-sdk-php#115]
When using the beta `/users/{email}/calendar/getSchedule` endpoint, the return logic of `ScheduleInformation#getScheduleItems` is wrong. 

```PHP
if (array_key_exists("scheduleItems", $this->_propDict)) {
    if (is_a($this->_propDict["scheduleItems"], "Microsoft\Graph\Beta\Model\ScheduleItem")) {
        return $this->_propDict["scheduleItems"];
    } else {
        $this->_propDict["scheduleItems"] = new ScheduleItem($this->_propDict["scheduleItems"]);
        return $this->_propDict["scheduleItems"];
    }
}
```
This should be iterating over the collection of `ScheduleItem` within `$this->_propDict["scheduleItems"]` and return an array of `ScheduleItem`.

Currently it is passing the whole collection to the `ScheduleItem` constructor which is causing it to return a general `array`. 
[AB#7578](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7578)

**Comment by MIchaelMainer** [comment]
This is an issue in the [PHP complexType template](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/blob/dev/Templates/PHP/Model/ComplexType.php.tt#L59). At first glance, it appears that our check of whether the property type is complex is incorrect. Additionally, when we have collections of objects, we should be returning the collection and not an array. This change would be breaking so we won't be fixing this until we are ready to have breaking moment PHP party.

**Comment by ngrie** [comment]
Any updates on this? Maybe there should at least exist an issue in the code generator repo?

**Comment by Blacksmoke16** [comment]
FWIW, I have since migrated my primary Github account.  There isn't a way to transfer the owner of an issue so for posterity you can contact this account if needed for anything related to this issue.

**Comment by Ndiritu** [comment]
@Blacksmoke16 thanks for informing us and apologies for the long wait. This will be triaged and prioritised.

**Comment by MIchaelMainer** [comment]
related to #76 and #97

**Comment by Ndiritu** [comment]
Fixed in the v2 preview release https://github.com/microsoftgraph/msgraph-sdk-php/issues/681

**Comment by isvargasmsft** [comment]
Hello, 

Thank you very much for your patience. This issue is resolved in our [new SDK version 2.0.0-RC5](https://github.com/microsoftgraph/msgraph-sdk-php/issues/943). 

Please feel free to give it a try and share your feedback with us.

Kind regards, 

Isaac Vargas
